client/setec/store.go

```diff
@@ -303,9 +303,21 @@ func (s *Store) Refresh(ctx context.Context) error {
 	}
 }
 
-// Secret returns a fetcher for the named secret. It returns nil if name does
-// not correspond to one of the secrets known by s.
+// Secret returns a fetcher for the named secret.
+//
+// If s has lookups enabled, Secret returns nil for an unknown name.
+// Otherwise, Secret panics for an unknown name.
 func (s *Store) Secret(name string) Secret {
+	sec := s.secretOrNil(name)
+	if sec == nil && !s.allowLookup {
+		panic(fmt.Sprintf("secret %q not found with lookup disabled", name))
```

Comment:
```suggestion
		panic(fmt.Sprintf("secret %q not found in StoreConfig.Secrets with lookup disabled", name))
```

Reply:
Good call. But I think maybe just `StoreConfig`, since it includes both Secrets and Structs, and then it gets a bit long 😆